client.go

```diff
@@ -25,6 +34,77 @@ func Dial(network, address string) (*Client, error) {
 	return NewClient(conn)
 }
 
+// DialConfig is used to pass configuration to DialURI()
+type DialConfig struct {
+	DTLSConfig dtls.Config
+	TLSConfig  tls.Config
+
+	Net transport.Net
+}
+
+// DialURI connect to the STUN/TURN URI and then
+// initializes Client on that connection, returning error if any.
+func DialURI(uri *URI, cfg *DialConfig) (*Client, error) {
+	var conn Connection
+	var err error
```

Comment:
better to just declare/initialize error when/where needed in the appropriate scopes below. Also an inner helper function can help get around the issue of wanting to assign conn and instead just return the conn, err and then construct a client or return an error

Reply:
I've moved the err declaration to where they are needed.